packages/@aws-cdk/aws-ecs/lib/container-image.ts

```diff
@@ -33,12 +44,22 @@ export abstract class ContainerImage {
    */
   public abstract readonly imageName: string;
 
+  /**
+   * Optional credentials for a private image registry
+   */
+  protected abstract readonly credentials?: RepositoryCreds;
+
   /**
    * Called when the image is used by a ContainerDefinition
    */
   public abstract bind(containerDefinition: ContainerDefinition): void;
+
+  /**
+   * Render the Repository credentials to the CloudFormation object
+   */
+  public abstract renderRepositoryCredentials(): CfnTaskDefinition.RepositoryCredentialsProperty | undefined;
```

Comment:
Usually we call these methods `toRepositoryCredentialsJson` (JSON indicates it's low-level)